File: packages/journey/android/src/main/java/com/pingidentity/rnjourney/JourneyErrorCodes.kt
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.pingidentity.rnjourney

/**
 * Stable Journey error code constants surfaced to JavaScript.
 */
internal object JourneyErrorCodes {
    const val CONFIG = "JOURNEY_CONFIG_ERROR"
    const val INIT = "JOURNEY_INIT_ERROR"
    const val START = "JOURNEY_START_ERROR"
    const val NEXT = "JOURNEY_NEXT_ERROR"
    const val RESUME = "JOURNEY_RESUME_ERROR"
    const val USER = "JOURNEY_USER_ERROR"
    const val LOGOUT = "JOURNEY_LOGOUT_ERROR"
    const val DISPOSE = "JOURNEY_DISPOSE_ERROR"
    const val STATE = "JOURNEY_STATE_ERROR"
    const val CALLBACK_APPLY = "JOURNEY_CALLBACK_APPLY_ERROR"
    const val UNSUPPORTED_CALLBACK = "JOURNEY_UNSUPPORTED_CALLBACK_ERROR"
    const val MISSING_INTEGRATION = "JOURNEY_MISSING_INTEGRATION_ERROR"
}

File: packages/journey/ios/Error/JourneyErrorCodes.swift
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import Foundation

/// Stable Journey error codes surfaced by iOS bridge rejections.
///
/// Keep these in sync with:
/// - JS `JourneyErrorCode` union (`packages/journey/src/types/error.types.ts`)
/// - Android `JourneyErrorCodes`.
enum JourneyErrorCodes: String {
  case configError = "JOURNEY_CONFIG_ERROR"
  case initError = "JOURNEY_INIT_ERROR"
  case startError = "JOURNEY_START_ERROR"
  case nextError = "JOURNEY_NEXT_ERROR"
  case resumeError = "JOURNEY_RESUME_ERROR"
  case userError = "JOURNEY_USER_ERROR"
  case logoutError = "JOURNEY_LOGOUT_ERROR"
  case disposeError = "JOURNEY_DISPOSE_ERROR"
  case stateError = "JOURNEY_STATE_ERROR"
  case callbackApplyError = "JOURNEY_CALLBACK_APPLY_ERROR"
  case unsupportedCallbackError = "JOURNEY_UNSUPPORTED_CALLBACK_ERROR"
  case missingIntegrationError = "JOURNEY_MISSING_INTEGRATION_ERROR"
}

